### DISCUSSION TITLE
Android app <---IKEv2---> Watchguard firewall

### DISCUSSION THREAD

**Discussion by thermseekr** [strongswan/strongswan#1453] [General]
Hello,

I'm trying to connect an IKEv2 VPN from an Android 11 phone to a Watchguard XTM5 series firewall. I already tried importing the profile created by the FW and also the manual config using its generated certificate, but can't get it to connect. The same settings work fine from an iPhone running iOS16.

I'm sorry, I have very limited knowledge about VPNs and its configuration. This is what I'm getting:

Android app:
```
Dec  9 12:20:46 00[DMN] +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
Dec  9 12:20:46 00[DMN] Starting IKE service (strongSwan 5.9.3rc1, Android 11 - RPAS31.Q2-59-17-4-3-9/2022-09-01, moto g(9) plus - motorola/odessa_retail/motorola, Linux 4.14.190-perf+, aarch64)
Dec  9 12:20:46 00[LIB] loaded plugins: androidbridge charon android-log openssl fips-prf random nonce pubkey chapoly curve25519 pkcs1 pkcs8 pem xcbc hmac socket-default revocation eap-identity eap-mschapv2 eap-md5 eap-gtc eap-tls x509
Dec  9 12:20:46 00[JOB] spawning 16 worker threads
Dec  9 12:20:46 06[IKE] initiating IKE_SA android[1] to xxx.xxx.xxx.xxx
Dec  9 12:20:46 06[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Dec  9 12:20:46 06[NET] sending packet: from yyy.yyy.yyy.yyy[37444] to xxx.xxx.xxx.xxx[500] (716 bytes)
Dec  9 12:20:46 09[NET] received packet: from xxx.xxx.xxx.xxx[500] to yyy.yyy.yyy.yyy[37444] (36 bytes)
Dec  9 12:20:46 09[ENC] parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
Dec  9 12:20:46 09[IKE] received NO_PROPOSAL_CHOSEN notify error
```

Watchguard XTM:
`
2022-12-09 13:20:47 iked (xxx.xxx.xxx.xxx<->yyy.yyy.yyy.yyy)IKEv2 IKE_SA_INIT exchange from yyy.yyy.yyy.yyy:31124 to xxx.xxx.xxx.xxx:500 failed. Gateway-Endpoint='WG Default IKEv2 Gateway'. Reason=IKE proposal did not match. Received DH group 19, expected 2.
`

The firewall is on a valid/public/static IP address (xxx) and the phone is on the operator's 4G network (yyy). I also tried from different LANs with same results.

Can you please provide me some pointers to try to solve this?

Thanks a lot
Tales

**Comment by Thermi** [comment]
Change the IKE proposal with the advanced settings

**Comment by thermseekr** [comment]
It worked! I struggled a bit navigating through the algorithm identifiers, until I finally figured it out. Thank you very much!
